File: app/src/main/java/com/movie/popcornapp/activities/moviedetails/MovieDetailsFragment.java
package com.movie.popcornapp.activities.moviedetails;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import com.movie.popcornapp.R;
import com.movie.popcornapp.databinding.FragmentMovieDetailsBinding;
import com.movie.popcornapp.infrastructure.factories.ViewModelFactory;
import com.movie.popcornapp.models.API.response.MovieResponse;

import java.util.ArrayList;

/**
 * @author george.radu on 2019-07-08.
 */
public class MovieDetailsFragment extends Fragment {

    //region Properties
    private FragmentMovieDetailsBinding binding;
    //endregion

    //region Lifecycle
    public MovieDetailsFragment() {
    }

    static MovieDetailsFragment newInstance(ArrayList<MovieResponse> selectedMovie) {
        MovieDetailsFragment fragment = new MovieDetailsFragment();
        Bundle args = new Bundle();
        args.putParcelableArrayList(MovieDetailsActivity.MOVIE_DETAILS_FRAGMENT_SELECTED_MOVIE_KEY, selectedMovie);
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_movie_details, null, false);
        return binding.getRoot();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        // get arguments and inject in xml
        MovieResponse selectedMovie = null;
        if (getArguments() != null) {
            selectedMovie = (MovieResponse) getArguments().getParcelableArrayList(MovieDetailsActivity.MOVIE_DETAILS_FRAGMENT_SELECTED_MOVIE_KEY).get(0);
        }

        MovieDetailsViewModel viewModel = ViewModelProviders.of(this, new ViewModelFactory(null, selectedMovie)).get(MovieDetailsViewModel.class);
        binding.setViewModel(viewModel);
    }
    //endregion
}

File: app/src/main/java/com/movie/popcornapp/activities/moviedetails/MovieDetailsFragment.kt
package com.movie.popcornapp.activities.moviedetails

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders
import com.movie.popcornapp.databinding.FragmentMovieDetailsBinding
import com.movie.popcornapp.infrastructure.factories.ViewModelFactory
import com.movie.popcornapp.models.API.response.MovieResponse
import java.util.ArrayList

class MovieDetailsFragment : Fragment() {

    //region Properties
    private var binding: FragmentMovieDetailsBinding? = null
    //endregion

    companion object {
        const val MOVIE_DETAILS_FRAGMENT_SELECTED_MOVIE_KEY = "movie_details_fragment_selected_movie_key"

        fun newInstance(selectedMovie: ArrayList<MovieResponse>): MovieDetailsFragment {
            val fragment = MovieDetailsFragment()
            val args = Bundle()
            args.putParcelableArrayList(MOVIE_DETAILS_FRAGMENT_SELECTED_MOVIE_KEY, selectedMovie)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return FragmentMovieDetailsBinding.inflate(inflater, container, false).also {
            binding = it
        }.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        // get arguments and inject in xml
        arguments?.getParcelableArrayList<MovieResponse>(MOVIE_DETAILS_FRAGMENT_SELECTED_MOVIE_KEY)!![0].also {
            ViewModelProviders.of(this, ViewModelFactory(null, it)).get(MovieDetailsViewModel::class.java).apply {
                binding?.viewModel = this
            }
        }
    }
    //endregion
}